### PR TITLE
Cephfs subvolume authorize test

### PR DESCRIPTION
**Pull request by yogesh-mane**
Description:
Added Cephfs subvolume authorize test - https://polarion.engineering.redhat.com/polarion/#/project/CEPH/workitem?id=CEPH-83574596

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs

# Checklist:

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarin Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test

### CODE REVIEW

**Comment by yogesh-mane** [comment]
5.x logs - http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-79PV6H

File: tests/cephfs/subvolume_authorize.py [hunk 1]
```diff
@@ -0,0 +1,372 @@
...
+import string
+import traceback
+
+from ceph.ceph import CommandFailed
+from tests.cephfs.cephfs_utilsV1 import FsUtils
+from tests.cephfs.client_authorize import verify_write_failure
+
+logger = logging.getLogger(__name__)
+log = logger
+
```
Comment (AmarnatReddy) [inline tests/cephfs/subvolume_authorize.py:12]: HI @yogesh-mane,
Can you add Doc-string for this test with details of what we are testing if possible

Comment (yogesh-mane) [reply tests/cephfs/subvolume_authorize.py:12]: Sure @AmarnatReddy

File: tests/cephfs/subvolume_authorize.py [hunk 2]
```diff
@@ -0,0 +1,426 @@
...
+                f"dd if=/dev/zero of={fuse_mount_dir}/file bs=10M count=10",
+                f"dd if={fuse_mount_dir}/file of={fuse_mount_dir}/file2 bs=10M count=10",
+            ]
+            for command in commands:
+                _, err = client[0].exec_command(
+                    sudo=True, cmd=command, long_running=True
+                )
+                if err:
+                    log.error(f"Permissions set for {client_name} is not working")
+                    return 1
```
Comment (psathyan) [inline tests/cephfs/subvolume_authorize.py:248]: I believe we have used the same steps in another test module. Can we implement a method and reuse it here?

File: tests/cephfs/cephfs_utilsV1.py [hunk 3]
```diff
@@ -815,3 +815,20 @@ def get_clone_status(self, client, vol_name, clone_name, **kwargs):
...
+            subvol_name: Cephfs subvolume name
+            client_name: Name of client to be created
+            **kwargs:
+                extra_params : subvolumegroup name, access level etc.
+        """
+        command = f"ceph fs subvolume authorize {vol_name} {subvol_name} {client_name}"
+        if kwargs.get("extra_params"):
+            command += f" {kwargs.get('extra_params')}"
+        out, rc = client.exec_command(sudo=True, cmd=command)
+        return 0
```
[the remark this answers is not in the document]
Comment (yogesh-mane) [reply tests/cephfs/cephfs_utilsV1.py:834]: Okay

File: tests/cephfs/subvolume_authorize.py [hunk 4]
```diff
@@ -0,0 +1,426 @@
...
+            log.info(traceback.format_exc())
+            return 1
+    except Exception as e:
+        log.info(e)
+        log.info(traceback.format_exc())
+        return 1
+    else:
+        log.error(f"Permissions set for {client_name} is not working for fuse mount")
+        return 1
+
```
[the remark this answers is not in the document]
Comment (yogesh-mane) [reply tests/cephfs/subvolume_authorize.py:70]: Okay

File: tests/cephfs/subvolume_authorize.py [hunk 5]
```diff
@@ -0,0 +1,426 @@
...
+        ceph fs subvolumegroup create <vol_name> <group_name> --pool_layout <data_pool_name>
+    28. Create cephfs subvolume in subvolumegroup
+        ceph fs subvolume create <vol_name> <subvol_name> [--size <size_in_bytes>] [--group_name <subvol_group_name>]
+    29. Reapeat steps 2-27 on cephfs subvolume created in above step (step 28)
+
+    Clean-up:
+    1. Remove all the data in Cephfs file system
+    2. ceph fs subvolume rm <vol_name> <subvol_name> [--group_name <subvol_group_name>]
+    3. ceph fs subvolumegroup rm <vol_name> <group_name>
+    4. Remove all the cephfs mounts
```
[the remark this answers is not in the document]
Comment (yogesh-mane) [reply tests/cephfs/subvolume_authorize.py:154]: Thanks

File: tests/cephfs/subvolume_authorize.py (hunk 2 above)
Comment (yogesh-mane) [reply tests/cephfs/subvolume_authorize.py:248]: Sure

**Comment by yogesh-mane** [comment]
subvolume_authorize test (4.x) - http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-ZCK9Z9
subvolume_authorize test (5.x) - http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-LJI97G

client_authorize test (4.x) - http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-XQKXVZ
client_authorize test (5.x) - http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-FVTS8R

I will post full run logs once  run is completed

**Comment by yogesh-mane** [comment]
5.x logs - http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-RW8Y4C
4.x logs - http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-T900AV

**Review by AmarnatReddy — APPROVED** [review]
LTGM

**Review by psathyan — APPROVED** [review]

**Pull request merged**